src/main/java/dinkplugin/util/KillCountService.java

```diff
@@ -0,0 +1,177 @@
+package dinkplugin.util;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import dinkplugin.notifiers.KillCountNotifier;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.NPC;
+import net.runelite.api.NpcID;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.events.NpcLootReceived;
+import net.runelite.client.plugins.chatcommands.ChatCommandsPlugin;
+import net.runelite.client.plugins.loottracker.LootReceived;
+import net.runelite.client.plugins.loottracker.LootTrackerConfig;
+import net.runelite.client.plugins.loottracker.LootTrackerPlugin;
+import net.runelite.http.api.loottracker.LootRecordType;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Singleton
+public class KillCountService {
+
+    private static final String RL_CHAT_CMD_PLUGIN_NAME = ChatCommandsPlugin.class.getSimpleName().toLowerCase();
+    private static final String RL_LOOT_PLUGIN_NAME = LootTrackerPlugin.class.getSimpleName().toLowerCase();
+
+    @Inject
+    private ConfigManager configManager;
+
+    @Inject
+    private Gson gson;
+
+    @Inject
+    private ScheduledExecutorService executor;
+
+    private final Cache<String, Integer> killCounts = CacheBuilder.newBuilder()
+        .expireAfterAccess(10, TimeUnit.MINUTES)
+        .build();
+
+    public void reset() {
+        this.killCounts.invalidateAll();
+    }
+
+    public void onNpcKill(NpcLootReceived event) {
+        NPC npc = event.getNpc();
+        int id = npc.getId();
+        if (id == NpcID.THE_WHISPERER || id == NpcID.THE_WHISPERER_12205 || id == NpcID.THE_WHISPERER_12206 || id == NpcID.THE_WHISPERER_12207) {
+            // Upstream does not fire NpcLootReceived for the whisperer, since they do not hold a reference to the NPC.
+            // So, we use LootReceived instead (and return here just in case they change their implementation).
+            return;
+        }
+
+        String name = npc.getName();
+        if (name != null) {
+            this.incrementKillCount(LootRecordType.NPC, name);
+        }
+    }
+
+    public void onLoot(LootReceived event) {
+        boolean increment;
+        if (event.getType() == LootRecordType.NPC && "The Whisperer".equalsIgnoreCase(event.getName())) {
+            // Special case: upstream fires LootReceived for the whisperer, but not NpcLootReceived
+            increment = true;
+        } else {
+            increment = event.getType() == LootRecordType.EVENT;
+        }
```

Comment:
code style question:
Could this be a switch instead?

```
boolean increment = false;
switch (event.getType()) {
  case LootRecordType.NPC:
    increment = "The Whisperer".equalsIgnoreCase(event.getName());
    break;
  case LootRecordType.Event:
    increment = true;
    break;
}
```

I don't know if that's very un-java, but that reads easier for me